http_disk_server.py

```diff
@@ -272,6 +278,13 @@ def parse_content_range(self):
             self.send_response(416)
             self.end_headers()
 
+        def log_message(self, format, *args):
+            dprint('%s - - [%s] %s\n' % (
+                self.address_string() if sys.version_info > (3,) else self.client_address[0],
+                self.log_date_time_string(),
+                format % args
+            ))
```

Comment:
Can you explain why this change is required?

Reply:
Part of the logs is generated by Python's HTTP server module. In order to control these logs, it is necessary to override `log_message` in the `RequestHandler`.

With something like this to provide the same output as the base function for both Python 2 and 3.

Reply:
The changes seem good to me, I wasn't very clear, I just wanted to know what logs it managed exactly so that it requires putting everything in debug. :)

Reply:
Oh ok sorry. This function manages all the logs from `HTTPServer`, like request logs `05/Feb/2025 09:49:46] "GET / HTTP/1.1" 206 -`, debug info `CGI script exited OK` and error messages like `Request timed out: %r` or CGI / setup related errors.

If it's too many logs being put in debug it is possible to override `log_request` instead. It is also possible to override `log_error` too in order to set log error messages with `eprint` instead.

Thinking about it, this might be filtering too many messages, if so I will adjust.

Reply:
Ok so, it seems good to override `log_request`.
Maybe something like that:
```python
        def _log_message(self, func, format, *args):
            func('%s - - [%s] %s\n' % (
                self.address_string() if sys.version_info > (3,) else self.client_address[0],
                self.log_date_time_string(),
                format % args
            ))

        def log_message(self, format, *args):
            self._log_message(dprint, format, *args)

        def log_error(self, format, *args):
            self._log_message(eprint, format, *args)
```

Reply:
done in [5191100](https://github.com/xcp-ng/http-nbd-transfer/pull/14/commits/5191100ac2a209d9b5df440d3cbe376bf3eee612)